.github/workflows/build.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and test
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  create:
+
+jobs:
+  build-and-test-code:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      
+      - name: create checksum file
+        run: find . -type f -name "*.gradle*" -o -name "gradle-wrapper*" | xargs shasum | sort > checksum.txt && cat checksum.txt
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.head_ref }}-${{ hashFiles('**/checksum.txt') }}
+
+      - name: download dependencies
+        run: ./gradlew downloadDependencies
+        
+      - name: Build with Gradle
+        run: ./gradlew build dockerBuildImages
+
+      - name: Unit test
+        run: ./gradlew jacocoTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          name: codecov-line-smoothing
+          fail_ci_if_error: true
+          verbose: true
+          flags: unit
+
+      - name: Integration test
+        run: ./gradlew jacocoIntegrationTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
```

Comment:
Will this upload the unit coverage as a duplicate under the integration flag? That's something we ran into before - I believe the circleci build has a clean in it to prevent that.

Reply:
Nope. it's not. 
check out this:
- Unit test: https://codecov.io/api/gh/hypertrace/attribute-service/download/build?path=v4/raw/2020-12-14/2EE1429A87F577B0E1D33CD9DDC07EE2/e0a6248fe64c0323f14fb2c7c102a41ce3416766/02f43c29-d412-4f68-9a4f-314f8aabb9ec.txt
- Integration test: https://codecov.io/api/gh/hypertrace/attribute-service/download/build?path=v4/raw/2020-12-14/2EE1429A87F577B0E1D33CD9DDC07EE2/e0a6248fe64c0323f14fb2c7c102a41ce3416766/dd7ed5b9-4b0c-40b8-afbd-dd6b11ec6355.txt